### PR TITLE
66 config editor

### PR DESCRIPTION
**Pull request by hogsy**
Merge for #66, not completed yet but good enough as a starting point - won't go any further with it for now.

If you're happy with it, please just merge it straight over rather than waiting on me.

### CODE REVIEW

File: src/engine/engine.cpp [hunk 1]
```diff
@@ -44,6 +44,12 @@ const char *GetCampaignPath(void) {
     return cv_campaign_path->s_value;
 }
 
+const char *GetFullCampaignPath(void) {
```
Comment (solemnwarning) [inline src/engine/engine.cpp]: Suggest returning std::string to avoid need for static buffer

File: src/engine/imgui_layer.cpp [hunk 2]
```diff
@@ -415,37 +444,44 @@ void AddFilePath(const char *path) {
...
-    } else if(
-            pl_strncasecmp(ext, "pps", 3) == 0
-            ) {
-        descriptor.type = FILE_TYPE_PARTICLE;
-    } else if(
-            pl_strncasecmp(ext, "wav", 3) == 0
-            ) {
-        descriptor.type = FILE_TYPE_AUDIO;
+    if(ext != nullptr) {
+        if(pl_strncasecmp(ext, "map", 3) == 0) {
```
Comment (solemnwarning) [inline src/engine/imgui_layer.cpp]: Why strncasecmp? this will match "mapX" etc. Same below.

File: src/engine/imgui_layer.cpp [hunk 3]
```diff
@@ -222,16 +194,38 @@ void MapConfigEditor::Display() {
...
 
-void MapConfigEditor::OpenManifest(const std::string &name) {
+void MapConfigEditor::OpenManifest(const std::string &path) {
+    std::string name = plGetFileName(path.c_str());
+    name.erase(name.find(".map"));
     manifest_ = *MapManager::GetInstance()->GetManifest(name);
+
     SyncManifest();
+
+    strncpy(filename_buffer, path.c_str(), sizeof(filename_buffer));
```
Comment (solemnwarning) [inline src/engine/imgui_layer.cpp]: This won't terminate filename_buffer if path is too long. Would these class variables be best as C++ strings?

Comment (hogsy) [reply src/engine/imgui_layer.cpp]: Will fix it, but ImGui writes into that so it can't be std string.

File: src/engine/engine.cpp (hunk 1 above)
Comment (hogsy) [reply src/engine/engine.cpp]: Can't because we still have older C code, but it's a good point for later.

File: src/engine/imgui_layer.cpp (hunk 3 above)
Comment (hogsy) [reply src/engine/imgui_layer.cpp]: Can't as ImGui doesn't support filling std string.

**Pull request merged**